teuthology/task/internal.py

```diff
@@ -726,19 +727,20 @@ def vm_setup(ctx, config):
     Look for virtual machines and handle their initialization
     """
     all_tasks = [x.keys()[0] for x in ctx.config['tasks']]
-    need_chef = False
-    if 'chef' in all_tasks or 'kernel' in all_tasks:
-        need_chef = True
-    with parallel() as p:
-        editinfo = os.path.join(os.path.dirname(__file__),'edit_sudoers.sh')
+    need_ansible = False
+    if 'kernel' in all_tasks and 'ansible.cephlab' not in all_tasks:
```

Comment:
Initially I effectively had it as an 'or'. Then I noticed ``ansible`` was being run twice on all scheduled jobs (because it is inserted by ``teuthology-suite`` just like ``chef`` was). Then I realized we'd been running ``chef`` twice on all scheduled ``vps`` jobs!!